lr-cli/cortx_setup/commands/storage/config.py

```diff
@@ -259,205 +272,224 @@ def run(self, **kwargs):
+                            " without enclosure id. Please provide user, password,"
+                            " ip and port together to fetch the enclosure id from"
+                            " attached enclosure.")
+                        raise RuntimeError(
+                            'Incomplete set of arguments provided'
+                        )
+            if self.enclosure_id is not None:
+                if setup_type == "VM":
+                    self.logger.warning("This is VM")
+                    self.logger.warning(
+                        "Adding ip and port in confstore without validation"
                     )
-                else:
-                    self.logger.error(
-                        "Controller type needs to be 'virtual' for VM")
-                    raise ValueError("Incorrect argument value provided")
-
-            if not name and not storage_type and not controller_type and self.cvg_count == -1:
-                self.logger.error(
-                    "Please provide valid arguments,"
-                    " for VM only {name, type, cvg} arguments are accepted"
+                if setup_type == "HW" and not cred_validation:
+                    self.logger.warning(
+                        "Adding ip and port in confstore without validation"
+                        "To force the validation, please run: cortx_setup"
+                        " storage config --controller <type> --mode primary"
+                        " --ip <ip> --port <port> --user <user> --password"
+                        " <password>"
+                    )
+                self.update_pillar_and_conf('ip', ip)
+                self.update_pillar_and_conf('port', port)
+            else:
+                self.logger.exception(
+                    "Could not update ip and port without enclosure id"
+                    "Please provide user, password, ip & port together")
+                raise RuntimeError(
+                    'Incomplete set of arguments provided'
                 )
-                raise RuntimeError("Incomplete arguments provided")
-        else:
-            if user != None and password != None and ip != None and port != None:
-                # fetch enclosure serial/id
-                self.enclosure_id = EnclosureInfo(
-                                        ip,
-                                        user,
-                                        password,
-                                        port
-                                    ).fetch_enclosure_serial()
-                self.mode = "primary"
-
-                self.refresh_key_map()
-
-                # store enclosure_id
-                self.store_in_file()
-                self.update_pillar_and_conf('enclosure_id', self.enclosure_id)
 
-                # store user
+        if user is not None or password is not None:
+            if (user and not password) or (password and not user):
```

Comment:
Please write simple logic to check example
`if user is None or password is None:`
`Please provide 'user' and 'passowrd' together"`

Reply:
Addressed.